src/utils/transform-data.js

```diff
@@ -0,0 +1,206 @@
+  point.histogram.no = point.histogram['0'];
+  point.histogram.yes = point.histogram['1'];
+  delete point.histogram['0'];
+  delete point.histogram['1'];
+  delete point.histogram['2'];
+}
+
+export function proportionsToCounts(point) {
+  point.counts = {};
+  Object.keys(point.proportions).forEach((p) => {
+    point.counts[p] = point.proportions[p] * point.total_users;
+  });
+}
+
+export function toAudienceSize(point) {
+  point.audienceSize = point.total_users;
+}
+
+export const makeLabel = {
+  version(pt) {
+    pt.label = pt.version;
+  },
+  build_id(pt) {
+    if (pt.build_date) {
+      pt.label = buildDateStringToDate(pt.build_date.slice(0, 18));
+    } else {
+      // if no build_date, let's attempt to use build_id instead.
+      // This is the current flow for Firefox Desktop.
+      // Add a build_date field here if one does not exist.
+      pt.label = fullBuildIDToDate(pt.build_id);
+      pt.build_date = new Date(pt.label);
+    }
+  },
+};
+
+/* quantile view transform functions */
+
+export const responseHistogramToGraphicFormat = (
+  point,
+  keyTransform = (k) => +k
+) => {
+  // turn histogram to array of objects, sorted.
+  const formatted = Object.entries(point.histogram).map(([k, v]) => ({
+    bin: keyTransform(k),
+    value: v,
+  }));
+  formatted.sort((a, b) => {
+    if (a.key > b.key) return -1;
+    if (a.key <= b.key) return 1;
+    return 0;
+  });
+  point.histogram = formatted;
+};
+
+export function transformedPercentiles(point) {
+  // requires responseHistogramToGraphicFormat to be run first
+  point.transformedPercentiles = Object.entries(point.percentiles).reduce(
+    (acc, [bin, value]) => {
+      // eslint-disable-next-line no-param-reassign
+      acc[bin] = nearestBelow(
+        value,
+        point.histogram.map((hi) => hi.bin)
+      );
+      return acc;
+    },
+    {}
+  );
+}
+
+export const standardProportionTransformations = [
+  checkForHistogram,
+  checkForTotalUsers,
+  toAudienceSize,
+  addProportion,
+  proportionsToCounts,
+];
+
+export const standardQuantileTransformations = [
+  checkForHistogram,
+  checkForPercentiles,
+  checkForTotalUsers,
+  toAudienceSize,
+  responseHistogramToGraphicFormat,
+  transformedPercentiles,
+];
+
+export const transformAPIResponse = {
+  quantile(data, aggregationLevel) {
+    const tr = transform(
+      ...standardQuantileTransformations,
+      makeLabel[aggregationLevel]
+    );
+    const transformed = [...tr(data)];
+    transformed.sort(sortByKey('label'));
+    return transformed;
+  },
+  proportion(data, aggregationLevel, probeType) {
+    const tr = transform(
+      // remove unused fields for histogram-boolean probe types (firefox desktop)
+      probeType === 'histogram-boolean' && changeBooleanHistogramResponse,
+      ...standardProportionTransformations,
+      makeLabel[aggregationLevel]
+    );
+    const transformed = [...tr(data)];
+    transformed.sort(sortByKey('label'));
+    return transformed;
+  },
+};
```

Comment:
```
   94:3   error  Assignment to property of function parameter 'point'  no-param-reassign            
   99:3   error  Assignment to property of function parameter 'point'  no-param-reassign            
  100:3   error  Assignment to property of function parameter 'point'  no-param-reassign            
  101:10  error  Assignment to property of function parameter 'point'  no-param-reassign            
  102:10  error  Assignment to property of function parameter 'point'  no-param-reassign            
  103:10  error  Assignment to property of function parameter 'point'  no-param-reassign
  107:3   error  Assignment to property of function parameter 'point'  no-param-reassign
  109:5   error  Assignment to property of function parameter 'point'  no-param-reassign
  114:3   error  Assignment to property of function parameter 'point'  no-param-reassign
  119:5   error  Assignment to property of function parameter 'pt'     no-param-reassign
  123:7   error  Assignment to property of function parameter 'pt'     no-param-reassign
  128:7   error  Assignment to property of function parameter 'pt'     no-param-reassign
  129:7   error  Assignment to property of function parameter 'pt'     no-param-reassign
  150:3   error  Assignment to property of function parameter 'point'  no-param-reassign
  155:3   error  Assignment to property of function parameter 'point'  no-param-reassign
```

Reply:
I'll be honest, I dislike this rule. The point of `transform` is to mutate the input for immer. Happy to just turn off `no-param-resassign` in `transform-data.js` – any other ideas? I don't don't see the utility of creating complete copies of these objects and passing them since that's sort of the point of using immer – don't need to continually copy objects when doing these types of state changes.

Reply:
Maybe I'll rename all these to `draft` and wait for your PR to land

Reply:
I get that. I think this rule can be useful at times, but JavaScript is not a purely functional language. It doesn't make sense to enforce it here. I'm fine with disabling `no-param-reassign` or using `draft`.

I'm sorry for not being more judicious before adding that comment. I should have been more clear. When it comes to ESLint, and it makes sense, feel free to disable a rule.

Reply:
FWIW, it's not just you. Other people hate how this rule interacts with immer. They also recommend using `draft` or disabling ESLint on these types of lines.

https://github.com/immerjs/immer/issues/189